src/drivers/device-events/index.js

```diff
@@ -0,0 +1,89 @@
+import xs from 'xstream';
+import {tap, omit, merge, identity} from 'ramda';
+
+
+const deviceEvents = [];
+
+const shakeGesture = () => {
+
+	if (!window.DeviceMotionEvent) {
+		return;
+	}
+
+
+
+	let last = null;
+
+	const threshold = 2;
+	const timeout = 1000;
+	let lastTime = new Date();
+
+	var query = window.matchMedia("(orientation:landscape)");
+	const coord = query.matches ? "x" : "y";
+
+	const deviceMotionHandler = eventData => {
+		let info = null;
+		var current = eventData.accelerationIncludingGravity;
+		var currentTime;
+		var timeDifference;
+
+		if (last === null) {
+			last = current.y;
+			return
+		}
+
+		const delta = Math.abs(last - current[coord]);
+
+		if (delta > threshold) {
+			currentTime = new Date();
+			timeDifference = currentTime.getTime() - lastTime.getTime();
+
+			if (timeDifference > timeout) {
+				deviceEvents.push({type: 'vertical-shake'});
+				lastTime = new Date();
+			}
+		}
+
+		last = current[coord];
+
+
+	};
+	window.addEventListener('devicemotion', deviceMotionHandler, false)
+};
+
+let streamInterval = 0;
+
+const deviceEventsProducer = {
```

Comment:
nice thanks, that actually was our confusion too

On Tue, Aug 2, 2016 at 2:06 PM László Vadász <notifications@github.com>
wrote:

> In src/drivers/device-events/index.js
> <https://github.com/maxinteger/iv-player/pull/4#discussion_r73142408>:
>
> > +			if (timeDifference > timeout) {
> > +				deviceEvents.push({type: 'vertical-shake'});
> > +				lastTime = new Date();
> > +			}
> > +		}
> > +
> > +		last = current[coord];
> > +
> > +
> > +	};
> > +	window.addEventListener('devicemotion', deviceMotionHandler, false)
> > +};
> > +
> > +let streamInterval = 0;
> > +
> > +const deviceEventsProducer = {
>
> Sorry, but I do not understand this part. Why you need this producer?
> You can create a stream from the devicemotion event and then you can map
> on it and do any calculation.
> If the browser trigger too oftern you can compose your stream with a
> debounce
>
> You can found the fromEvent and the debounce in the XStream extra folder
> https://github.com/staltz/xstream/blob/master/src/extra
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/maxinteger/iv-player/pull/4/files/bb7121148f112deaeab3781d6e59aaf8bb270016#r73142408>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB2erhc6r6YZXWv-2_MwTQezkmHW5ifzks5qbzLWgaJpZM4JahwM>
> .
>
-- 
György Fekete
Senior Developer @ Pulilab

E-mail: f@pulilab.com f@vidzor.com
Web: pulilab.com vidzor.com